src/nvim/tui/tui.c

```diff
@@ -1864,9 +1866,6 @@ void tui_guess_size(TUIData *tui)
 
   tui_set_size(tui, width, height);
 
-  // Redraw on SIGWINCH event if size didn't change. #23411
-  ui_client_set_size(width, height);
```

Comment:
https://github.com/neovim/neovim/pull/23415 added tests for #23411, which still pass in this PR 👍